client/src/components/Datalist/index.js

```diff
@@ -4,17 +4,19 @@ import { uniq } from "lodash";
 class Datalist extends React.PureComponent {
   render() {
     let options = [];
-
     if (this.props.options && this.props.processOptions) {
-      options = this.props.processOptions(this.props.options);
+      options = this.props.processOptions(
+        this.props.options,
+        this.props.optionalArgs
+      );
     } else if (this.props.options) {
       options = this.props.options;
     }
 
     return (
       <datalist id={this.props.id}>
         {uniq(options).map(option => (
-          <option key={option} value={option}>
+          <option key={!option ? "" : option} value={option}>
```

Comment:
Try to not have double negative conditionals. e.g., this would be better: `option ? option : ""`

But this can be simplified further with a short-circuit boolean: `option || ""`

That said... why would `option` ever be falsy when mapping through them? Why not just `option`?

Reply:
This should all become moot if you revert the changes to Datalist, but worth mentioning that the code above would potentially allow multiple options with the same key, which defeats the purpose of using the `key` prop.

Reply:
i fixed this up so it's just `option || ""`. option is falsy when there hasn't been a category selected yet. as for the unique keys, i've swapped the lodash `uniq` (which would prevent duplicates) with a Set since that's the only reason lodash was being included.

Reply:
I don't understand how using `uniq` would help you here.  You're saying that if `option` is falsey, the key should be set as an empty string.  Presumably you could wind up in a situation where multiple options have evaluated to falsey and you now how have multiple option elements with the same key (an empty string). 
In any case, please avoid changing this file at all and alter the `options` and `processOptions` props you're passing to be used as intended.  Actually, with your refactor you shouldn't need to supply a `processOptions` param at all to your datalist.  Just do your processing in `TeaCollectionTableContainer` and pass the options as an array of strings to the `Datalist` component.